examples/step-69/step-69.cc

```diff
@@ -0,0 +1,2926 @@
+  //
+  // Finally we note that:
+  //  - This is the only class in this tutorial step that is tied to a
+  //    particular "physics" or "hyperbolic conservation law" (in this
+  //    case Euler's equations). All the other classes are primarily
+  //    "discretization" classes, very much agnostic of the particular physics
+  //    being solved.
+  //  - This is a "pure static" class (the antithesis of a
+  //    "pure virtual" class). It's just a convenient way to wrap-up a
+  //    collection of related methods into a single object. Note that we will
+  //    be able to invoke such methods without without creating an instance of
+  //    the class. Similarly, we will not have to provide a constructor
+  //    for this class.
+
+  template <int dim>
+  class ProblemDescription
+  {
+  public:
+    /* constexpr tells the compiler to evaluate "2 + dim" just once at compile
+       time rather than everytime problem_dimension is invoked. */
+    static constexpr unsigned int problem_dimension = 2 + dim;
+
+    using rank1_type = Tensor<1, problem_dimension>;
+    using rank2_type = Tensor<1, problem_dimension, Tensor<1, dim>>;
+
+    const static std::array<std::string, dim + 2> component_names;
+
+    static constexpr double gamma = 7. / 5.;
+
+    static DEAL_II_ALWAYS_INLINE inline Tensor<1, dim>
+    momentum(const rank1_type &U);
+
+    static DEAL_II_ALWAYS_INLINE inline double
+    internal_energy(const rank1_type &U);
+
+    static DEAL_II_ALWAYS_INLINE inline double pressure(const rank1_type &U);
+
+    static DEAL_II_ALWAYS_INLINE inline double
+    speed_of_sound(const rank1_type &U);
+
+    static DEAL_II_ALWAYS_INLINE inline rank2_type f(const rank1_type &U);
+
+    static DEAL_II_ALWAYS_INLINE inline double
+    compute_lambda_max(const rank1_type &    U_i,
+                       const rank1_type &    U_j,
+                       const Tensor<1, dim> &n_ij);
+  };
+
+  // @sect4{The <code>InitialValues</code> class}
+  //
+  // The class <code>InitialValues</code>'s only public data attribute is a
+  // std::function <code>initial_state</code> that computes the initial
+  // state of a given point and time. For the purpose of this example step
+  // we simply implement a homogeneous uniform flow field for which the
+  // direction and a 1D primitive state (density, velocity, pressure) are
+  // read from the parameter file.
+  //
+  // It would be desirable to initialize the class in a single shot:
+  // initialize/set the parameters and define the class members that
+  // depend on these default parameters. However, since we do not know the
+  // actual final values for the parameters, this would be sort of
+  // meaningless an unsafe in general (we would like to have mechanisms to
+  // check the consistency of the input parameters). Instead of defining
+  // another <code>setup()</code> method to be called (by-hand) after the
+  // call to <code> ParameterAcceptor::initialize() </code> we provide an
```

Comment:
doxygen will get it right:
```suggestion
  // call to ParameterAcceptor::initialize() we provide an
```